src/renderers/dom/stack/client/wrappers/ReactDOMInput.js

```diff
@@ -195,8 +195,12 @@ var ReactDOMInput = {
         // Simulate `input.valueAsNumber`. IE9 does not support it
         var valueAsNumber = parseFloat(node.value, 10) || 0;
 
-        // eslint-disable-next-line
-        if (value != valueAsNumber) {
+        if (
+          // eslint-disable-next-line
+          value != valueAsNumber ||
+          // eslint-disable-next-line
+          (value == valueAsNumber && node.value != value)
+        ) {
```

Comment:
This logic would have to be updated in the Fiber implementation as well. See [here](https://github.com/facebook/react/blob/master/src/renderers/dom/fiber/wrappers/ReactDOMFiberInput.js#L207). It's been updated since the alpha release you mentioned.